src/utils/Locales.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file Used to get the list of locales supported by Box
+ * @author Box
+ */
+import loadLocaleData from '@box/cldr-data';
+import { LocaleData, LanguagesData } from '@box/cldr-data/types';
+
+export interface BoxLanguage {
+    id: number;
+    name: string;
+}
+
+const DEFAULT_LOCALE = 'en-US';
+const DEFAULT_DISPLAY_NAMES: Array<BoxLanguage> = [
+    { id: 4, name: 'English (US)' },
+    { id: 59, name: 'English (UK)' },
+    { id: 68, name: 'Dansk' },
+    { id: 18, name: 'Deutsch' },
+    { id: 67, name: 'English (Australia)' },
+    { id: 66, name: 'English (Canada)' },
+    { id: 10, name: 'Español' },
+    { id: 72, name: 'Español (Latinoamérica)' },
+    { id: 21, name: 'Français' },
+    { id: 65, name: 'Français (Canada)' },
+    { id: 16, name: 'Italiano' },
+    { id: 24, name: 'Nederlands' },
+    { id: 69, name: 'Norsk (Bokmål)' },
+    { id: 30, name: 'Polski' },
+    { id: 14, name: 'Português' },
+    { id: 61, name: 'Suomi' },
+    { id: 57, name: 'Svenska' },
+    { id: 49, name: 'Türkçe' },
+    { id: 8, name: 'Русский' },
+    { id: 71, name: 'हिन्दी' },
+    { id: 70, name: 'বাংলা' },
+    { id: 19, name: '日本語' },
+    { id: 6, name: '简体中文' },
+    { id: 63, name: '繁體中文' },
+    { id: 55, name: '한국어' },
+];
+
+class Locales {
+    locale: string;
+
+    localeData: LocaleData;
+
+    languages: LanguagesData;
+
+    /**
+     * [constructor]
+     *
+     * @return {void}
+     */
+    constructor(locale?: string) {
+        this.locale = locale || DEFAULT_LOCALE;
+        this.localeData = loadLocaleData(this.locale);
+        this.languages = this.localeData.languages;
+    }
+
+    /**
+     * Compararator for language names
+     *
+     * @param  {BoxLanguage} locale1
+     * @param  {BoxLanguage} locale2
+     *
+     * @return {Array<string>}
+     */
+    localeComparator = (locale1: BoxLanguage, locale2: BoxLanguage) => locale1.name.localeCompare(locale2.name);
```

Comment:
This will not do a locale-sensitive sort in all browsers. You can try

```javascript
import { language } from '../elements/common/i18n';

localeComparator = (locale1: BoxLanguage, locale2: BoxLanguage) => locale1.name.localeCompare(locale2.name, language);
```

This will catch the more modern ones that support the Intl object and have the Intl.Collator support. Older ones that do not will ignore the "extra" parameter and just use English compare.